Mathlib/Analysis/Convex/Strong.lean

```diff
@@ -0,0 +1,180 @@
+  ⟨hf.1, fun x hx y hy a b ha hb hab ↦ (hf.2 hx hy ha hb hab).trans $
+    sub_le_sub_left (mul_le_mul_of_nonneg_left (hψφ _) $ by positivity) _⟩
+
+lemma UniformConcaveOn.mono (hψφ : ψ ≤ φ) (hf : UniformConcaveOn s φ f) : UniformConcaveOn s ψ f :=
+  ⟨hf.1, fun x hx y hy a b ha hb hab ↦ (hf.2 hx hy ha hb hab).trans' $
+    add_le_add_left (mul_le_mul_of_nonneg_left (hψφ _) $ by positivity) _⟩
+
+lemma UniformConvexOn.convexOn (hf : UniformConvexOn s φ f) (hφ : 0 ≤ φ) : ConvexOn ℝ s f := by
+  simpa using hf.mono hφ
+
+lemma UniformConcaveOn.concaveOn (hf : UniformConcaveOn s φ f) (hφ : 0 ≤ φ) : ConcaveOn ℝ s f := by
+  simpa using hf.mono hφ
+
+lemma UniformConvexOn.strictConvexOn (hf : UniformConvexOn s φ f) (hφ : ∀ r, r ≠ 0 → 0 < φ r) :
+    StrictConvexOn ℝ s f := by
+  refine ⟨hf.1, fun x hx y hy hxy a b ha hb hab ↦ (hf.2 hx hy ha.le hb.le hab).trans_lt $
+    sub_lt_self _ ?_⟩
+  rw [←sub_ne_zero, ←norm_pos_iff] at hxy
+  have := hφ _ hxy.ne'
+  positivity
+
+lemma UniformConcaveOn.strictConcaveOn (hf : UniformConcaveOn s φ f) (hφ : ∀ r, r ≠ 0 → 0 < φ r) :
+    StrictConcaveOn ℝ s f := by
+  refine ⟨hf.1, fun x hx y hy hxy a b ha hb hab ↦ (hf.2 hx hy ha.le hb.le hab).trans_lt' $
+    lt_add_of_pos_right _ ?_⟩
+  rw [←sub_ne_zero, ←norm_pos_iff] at hxy
+  have := hφ _ hxy.ne'
+  positivity
+
+lemma UniformConvexOn.add (hf : UniformConvexOn s φ f) (hg : UniformConvexOn s ψ g) :
+    UniformConvexOn s (φ + ψ) (f + g) := by
+  refine ⟨hf.1, fun x hx y hy a b ha hb hab ↦ ?_⟩
+  simpa [mul_add, add_add_add_comm, sub_add_sub_comm]
+    using add_le_add (hf.2 hx hy ha hb hab) (hg.2 hx hy ha hb hab)
+
+lemma UniformConcaveOn.add (hf : UniformConcaveOn s φ f) (hg : UniformConcaveOn s ψ g) :
+    UniformConcaveOn s (φ + ψ) (f + g) := by
+  refine ⟨hf.1, fun x hx y hy a b ha hb hab ↦ ?_⟩
+  simpa [mul_add, add_add_add_comm] using add_le_add (hf.2 hx hy ha hb hab) (hg.2 hx hy ha hb hab)
+
+lemma UniformConvexOn.neg (hf : UniformConvexOn s φ f) : UniformConcaveOn s φ (-f) := by
+  refine ⟨hf.1, fun x hx y hy a b ha hb hab ↦ le_of_neg_le_neg ?_⟩
+  simpa [add_comm, -neg_le_neg_iff, le_sub_iff_add_le'] using hf.2 hx hy ha hb hab
+
+lemma UniformConcaveOn.neg (hf : UniformConcaveOn s φ f) : UniformConvexOn s φ (-f) := by
+  refine ⟨hf.1, fun x hx y hy a b ha hb hab ↦ le_of_neg_le_neg ?_⟩
+  simpa [add_comm, -neg_le_neg_iff, ←le_sub_iff_add_le', sub_eq_add_neg, neg_add]
+    using hf.2 hx hy ha hb hab
+
+lemma UniformConvexOn.sub (hf : UniformConvexOn s φ f) (hg : UniformConcaveOn s ψ g) :
+    UniformConvexOn s (φ + ψ) (f - g) := by simpa using hf.add hg.neg
+
+lemma UniformConcaveOn.sub (hf : UniformConcaveOn s φ f) (hg : UniformConvexOn s ψ g) :
+    UniformConcaveOn s (φ + ψ) (f - g) := by simpa using hf.add hg.neg
+
+/-- A function `f` from a real normed space is `m`-strongly convex if
+`f (t • x + (1 - t) • y) ≤ t • f x + (1 - t) • f y - t * (1 - t) * φ ‖x - y‖` for all `t ∈ [0, 1]`.
```

Comment:
I think you mean "`x ↦ f x - m / 2 * ‖x‖ ^ 2` is convex" is the definition and it is equivalent to the uniformly convex in an inner product space.

Reply:
Your comment doesn't make sense, but so does my docstring! Fixed.